cloudinit/cmd/main.py

```diff
@@ -987,7 +987,11 @@ def main(sysv_args=None):
 
     if sysv_args:
         # Only load subparsers if subcommand is specified to avoid load cost
-        subcommand = sysv_args[0]
+        subcommand = list(
+            filter(lambda arg: not arg.startswith("-"), sysv_args)
+        )
+        if subcommand:
+            subcommand = subcommand[0]
```

Comment:
Let's just use next here as it'll be faster than the filter + lambda + list cast.
```suggestion
        subcommand = next(
            (posarg for posarg in sysv_args if not posarg.startswith("-")),
            None
        )
```

Reply:
Oh, this is a better approach.